mindsdb/integrations/handlers/s3_handler/s3_handler.py

```diff
@@ -17,129 +15,198 @@
 
+    def _connect_duckdb(self) -> DuckDBPyConnection:
+        """
+        Establishes a connection to the AWS (S3) account via DuckDB.
+
+        Returns:
+            DuckDBPyConnection: A connection object to the AWS (S3) account.
+        """
+        # Connect to S3 via DuckDB.
+        duckdb_conn = duckdb.connect()
+        duckdb_conn.execute("INSTALL httpfs")
+        duckdb_conn.execute("LOAD httpfs")
+
+        # Configure mandatory credentials.
+        duckdb_conn.execute(f"SET s3_access_key_id='{self.connection_data['aws_access_key_id']}'")
+        duckdb_conn.execute(f"SET s3_secret_access_key='{self.connection_data['aws_secret_access_key']}'")
+
+        # Configure optional parameters.
+        if 'aws_session_token' in self.connection_data:
+            duckdb_conn.execute(f"SET s3_session_token='{self.connection_data['aws_session_token']}'")
+
+        if 'region_name' in self.connection_data:
+            duckdb_conn.execute(f"SET s3_region='{self.connection_data['region_name']}'")
+
+        return duckdb_conn
+
+    def _connect_boto3(self) -> boto3.client:
+        """
+        Establishes a connection to the AWS (S3) account via boto3.
+
+        Returns:
+            boto3.client: A client object to the AWS (S3) account.
+        """
+        # Configure mandatory credentials.
+        config = {
+            'aws_access_key_id': self.connection_data['aws_access_key_id'],
+            'aws_secret_access_key': self.connection_data['aws_secret_access_key']
+        }
+
+        # Configure optional parameters.
+        if 'aws_session_token' in self.connection_data:
+            config['aws_session_token'] = self.connection_data['aws_session_token']
+
+        # DuckDB considers us-east-1 to be the default region.
+        config['region_name'] = self.connection_data['region_name'] if 'region_name' in self.connection_data else 'us-east-1'
+
+        return boto3.client('s3', **config)
+
     def disconnect(self):
-        """ Close any existing connections
-        Should switch self.is_connected.
         """
+        Closes the connection to the AWS (S3) account if it's currently open.
+        """
+        if not self.is_connected:
+            return
+        self.connection.close()
         self.is_connected = False
-        return
 
     def check_connection(self) -> StatusResponse:
         """
-        Check connection to the handler.
+        Checks the status of the connection to the S3 bucket.
+
         Returns:
-            HandlerStatusResponse
+            StatusResponse: An object containing the success status and an error message if an error occurs.
         """
-
         response = StatusResponse(False)
         need_to_close = self.is_connected is False
 
+        # Check connection via boto3.
         try:
-            connection = self.connect()
-            connection.head_object(Bucket=self.connection_data['bucket'], Key=self.connection_data['key'])
+            boto3_conn = self._connect_boto3()
```

Comment:
Can we somehow do this through Duckdb, or is this enough for now?

Reply:
@ZoranPandovski I noticed that even if we run something like `SELECT 1;` with incorrect credentials, it runs without an issue. I think the only way to reliably check the connection through DuckDB is to get a list of files in the bucket and query one of them. Do you think we should do this?